src/mmw/js/src/core/models.js

```diff
@@ -61,36 +61,89 @@ var TaskModel = Backbone.Model.extend({
         }
     },
 
+    // Cancels any currently running jobs. The promise returned
+    // by previous calls to pollForResults will be rejected.
+    reset: function() {
+        this.set({
+            'job': null,
+            'result': null,
+            'status': null
+        });
+    },
+
+    // taskHelper should be an object containing an optional object,
+    // postData, an optional function, onStart, and functions pollSuccess,
+    // pollFailure, and startFailure.
+    start: function(taskHelper) {
+        this.reset();
+        if (taskHelper.onStart) {
+            taskHelper.onStart();
+        }
+        var self = this;
+
+        self
+            .fetch({
+                method: 'POST',
+                data: taskHelper.postData
+            })
+            .done(function() {
+                self.pollForResults()
+                    .done(function() {
+                        taskHelper.pollSuccess();
```

Comment:
If this function is optional, we should check that it exists before calling it. Also consider making it a class method.